### PR TITLE
taxtweb

### PR DESCRIPTION
**Pull request by nastasi-oq**
taxtweb application included and extended to be used by vulnerability form as pop-up tool.
Currently not bidirectionality between form-field and pop-up , just monodirectional from pop-up to form field.
**NOTE**: your local_settings.py needs some change, do diff with the ```local_settings.py.template```.

### CODE REVIEW

File: openquakeplatform/openquakeplatform/taxtweb/urls.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# vim: tabstop=4 shiftwidth=4 softtabstop=4
+
+# Copyright (c) 2014, GEM Foundation.
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/taxtweb/urls.py]: Update the copyright

File: openquakeplatform/openquakeplatform/taxtweb/views.py [hunk 2]
```diff
@@ -0,0 +1,884 @@
+# Copyright (c) 2014, GEM Foundation.
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/taxtweb/views.py]: Update the copyright

File: openquakeplatform/openquakeplatform/taxtweb/views.py [hunk 3]
```diff
@@ -0,0 +1,884 @@
...
+    return render_to_response("taxtweb/index.html",
+                              dict(is_popup=is_popup,
+                                   type_of_assessment=type_of_assessment,
+                                   menu_content=menu_content,
+                                   sub1menu_content=sub1menu_content,
+                                   ),
+                              context_instance=RequestContext(request))
+
+
+# def list_entries(request, **kwargs):
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/taxtweb/views.py]: Is it ok to have all this code commented?

File: openquakeplatform/openquakeplatform/vulnerability/widgets.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Copyright (c) 2012-2013, GEM Foundation.
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/vulnerability/widgets.py]: Update the copyright

File: openquakeplatform/openquakeplatform/taxtweb/views.py (hunk 3 above)
Comment (nastasi-oq) [inline openquakeplatform/openquakeplatform/taxtweb/views.py]: Removing.

**Comment by daniviga** [comment]
After a complete reinstall I get an error editing a curve which looks related:

```python
Internal Server Error: /admin/vulnerability/generalinformation/7/
Traceback (most recent call last):
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 140, in get_response
    response = response.render()
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/response.py", line 105, in render
    self.content = self.rendered_content
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/response.py", line 82, in rendered_content
    content = template.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 140, in render
    return self._render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 830, in render
    bit = self.render_node(node, context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/debug.py", line 74, in render_node
    return node.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 124, in render
    return compiled_parent._render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 830, in render
    bit = self.render_node(node, context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/debug.py", line 74, in render_node
    return node.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 124, in render
    return compiled_parent._render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 830, in render
    bit = self.render_node(node, context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/debug.py", line 74, in render_node
    return node.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 830, in render
    bit = self.render_node(node, context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/debug.py", line 74, in render_node
    return node.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 63, in render
    result = block.nodelist.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 830, in render
    bit = self.render_node(node, context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/debug.py", line 74, in render_node
    return node.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/defaulttags.py", line 189, in render
    nodelist.append(node.render(context))
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 156, in render
    return self.render_template(self.template, context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/loader_tags.py", line 138, in render_template
    output = template.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 140, in render
    return self._render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 134, in _render
    return self.nodelist.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 830, in render
    bit = self.render_node(node, context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/debug.py", line 74, in render_node
    return node.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/defaulttags.py", line 189, in render
    nodelist.append(node.render(context))
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/defaulttags.py", line 189, in render
    nodelist.append(node.render(context))
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/defaulttags.py", line 285, in render
    return nodelist.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 830, in render
    bit = self.render_node(node, context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/debug.py", line 74, in render_node
    return node.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/defaulttags.py", line 285, in render
    return nodelist.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/base.py", line 830, in render
    bit = self.render_node(node, context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/debug.py", line 74, in render_node
    return node.render(context)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/template/debug.py", line 87, in render
    output = force_text(output)
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/utils/encoding.py", line 99, in force_text
    s = s.__unicode__()
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/forms/forms.py", line 411, in __str__
    return self.as_widget()
  File "/home/ubuntu/platform-env/local/lib/python2.7/site-packages/django/forms/forms.py", line 458, in as_widget
    return widget.render(name, self.value(), attrs=attrs)
  File "/home/ubuntu/oq-platform/openquakeplatform/openquakeplatform/vulnerability/widgets.py", line 38, in render
    final_attrs['value'] = force_text(self._format_value(value))
NameError: global name 'force_text' is not defined
[13/Jan/2015 16:02:32] "GET /admin/vulnerability/generalinformation/7/? HTTP/1.1" 500 776899
```

File: openquakeplatform/openquakeplatform/vulnerability/widgets.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from django.utils.html import format_html
+from django.forms.util import flatatt
```
Comment (daniviga) [inline openquakeplatform/openquakeplatform/vulnerability/widgets.py]: missing `force_text`: `from django.forms.util import flatatt, force_text`

**Comment by daniviga** [comment]
Probably an auto-resizing pop-up would be better (it's too narrow on my setup), but this can be addressed later. LGTM

**Pull request merged**